### ISSUE TITLE
SNOWFLAKE_COLUMNS_CASE_SENSITIVE may not be compatible with GraphQL

### ISSUE THREAD

**Issue by c-tanner** [appsfortableau/laravel-pdo-odbc#37]
When using Lighthouse, field name cases does not seem to be uppercased as the config value suggests it should be.

Snowflake schema has table `ACCOUNT` with `ID`,`NAME` columns.

Laravel model:

```php
class Account extends Model
{

    protected $connection = 'snowflake_pdo';
    protected $table = 'account';

}
```

GraphQL Schema:

```graphql
type Query {
  Accounts: [Account]! @all
}

type Account {
  id: Int!
  name: String
}
```

GraphQL query:

```graphql
query {
    Accounts {
        id
        name
    }
}
```

Error:

```
Cannot return null for non-nullable field "Account.id"
```

When fields in GraphQL schema are uppercased, the query executes successfully.

System specs:
```
laravel/framework                                  11.27.2
nuwave/lighthouse                                 6.45.0 
yoramdelangen/laravel-pdo-odbc        1.2.5
```

**Comment by c-tanner** [comment]
After more discovery, this seems to be due to the way that Lighthouse structures queries. Inspecting the `wrapColumn` function shows that even when specifying columns in the GraphQL query, the query actually calls `*`, bypassing the uppercasing.

**Comment by yoramdelangen** [comment]
Agree on resolving it nicely. There is a connection setting in Snowflake ODBC that allows you to control how Snowflake should handle the [double quoted wrapped column names](https://docs.snowflake.com/en/sql-reference/identifiers-syntax#migrating-from-databases-that-treat-double-quoted-identifiers-as-case-insensitive). 

By default we have set this setting to `false` so when double quoted we control the casing. You can disable it, by enabling `SNOWFLAKE_DISABLE_FORCE_QUOTED_IDENTIFIER=true`. You can find it [here](https://github.com/appsfortableau/laravel-pdo-odbc/blob/main/src/Flavours/Snowflake/Connector.php#L62-L64). 

Maybe this resolves the problem for you?

**Comment by c-tanner** [comment]
Thanks for the reply, unfortunately this doesn't solve the issue in this situation.

While the Laravel model attributes can now be lowercased (with caveats, below), the GraphQL schema definitions must still be uppercase in order for the query to execute successfully. 

What's interesting - model relationships do not return expected results when used through Lighthouse. When attributes are lowercased ,a query that calls relationships returns an empty array for the relationship. Single model queries work as expected.

To recap:

env:
```
SNOWFLAKE_COLUMNS_CASE_SENSITIVE=false
SNOWFLAKE_DISABLE_FORCE_QUOTED_IDENTIFIER=true
```

```graphql
"Working"
type Account {
  ID: Int!
  NAME: String
}
```

```graphql
"Not working"
type Account {
  id: Int!
  name: String
}
```

```php
# Working for single model queries
class Account extends Model
{
    protected $connection = 'snowflake_pdo';
    protected $primaryKey = 'id';
    protected $table = 'account';
}
```

```php
# Not working for queries calling relationships
class Account extends Model
{
    protected $connection = 'snowflake_pdo';
    protected $primaryKey = 'id';
    protected $table = 'account';

    public function children(): HasMany
    {
        return $this->hasMany(Children::class, 'parent_id');
    }
}
```

From the above example, this query would return an empty `children` array, with `Account` data being returned as expected:

```graphql
query {
    Accounts {
        ID
        NAME
        children {
            ID
            NAME
        }
    }
}
```

Works as expected when model attributes are all caps.

**Comment by yoramdelangen** [comment]
@c-tanner I though I had wrote a response alright.. but somehow it got lost :D

I know there is a server-sided option for Snowflake as well to set the casing:
https://docs.snowflake.com/en/sql-reference/parameters#quoted-identifiers-ignore-case

This option can also be set on session level, making the database connection string option obsolete? 

Another note: How does the transformation of the GraphQL query look like in SQL statement? Note that None quoted field names always be set to uppercased names. Check their ["Identifier requirements"](https://docs.snowflake.com/en/sql-reference/parameters#quoted-identifiers-ignore-case) documentation.

So in short:
```graphql
query {
    Accounts {
        id
        name
    }
}
```
Should transform into something like:
```sql
SELECT "id", "name" FROM Accounts
-- note that Account will be UPPERCASED as well
```
But instead it is currently transformed into:
```sql
SELECT id, name FROM Accounts
-- interpret by snowflake as:
SELECT ID, NAME FROM ACCOUNTS
```
